app/chat-history/singleCall.tsx

```diff
@@ -0,0 +1,188 @@
+import CallsCard from '@/components/chats/voiceCard';
+import { Text, View, Image, TouchableOpacity, Dimensions, Animated, Pressable } from 'react-native';
+import { phone } from '@/assets/icons/phone';
+import Svg, { Line, SvgXml } from 'react-native-svg';
+import { moreTransparent } from '@/assets/icons/more';
+import { back } from '@/assets/icons/userprofile/icons';
+import Button from '@/components/buttonModified';
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { play } from "@/assets/icons/playBtn";
+import { download } from '@/assets/icons/download';
+import { deleteRed } from '@/assets/icons/delete';
+import { router } from 'expo-router';
+import { AVPlaybackSource, AVPlaybackStatus, AVPlaybackStatusSuccess, Audio } from "expo-av";
+import { Sound } from 'expo-av/build/Audio';
+
+const { width } = Dimensions.get('window');
+const WAVEFORM_ELEMENTS_COUNT = 50;
+export default function SingleCall() {
+    const [hidden, setHidden] = useState(false);
+    const [visible, setVisible] = useState(false);
+    const [isPlaying, setIsPlaying] = useState(false);
+    const [position, setPosition] = useState<number>(0);
+    const [duration, setDuration] = useState<number | any>(1);
+    const [sound, setSound] = useState<Sound | null>(null);
+    const progressAnim = useRef(new Animated.Value(0)).current;
+    const intervalRef = useRef<NodeJS.Timer | null>(null);
+    const [waveformHeights, setWaveformHeights] = useState<number[]>([]);
+
+    const generateWaveformHeights = () => {
```

Comment:
I tried to look through many resources but in order the ways you can generate waves from an audio is to record that audio first by enabling metering which will allow you to capture real time noise level in decibels where you can use those level to interpolate the height of those waves, may be we should let this as it is I will continue to search for this, instead of using this whole time for this thing.